core/src/main/java/org/elasticsearch/index/snapshots/IndexShardSnapshotStatus.java

```
@@ -171,71 +149,110 @@ public synchronized void addProcessedFile(long size) {
     }
 
     /**
-     * Number of files
+     * Returns a copy of the current {@link IndexShardSnapshotStatus}. This method is
+     * intended to be used when a coherent state of {@link IndexShardSnapshotStatus} is needed.
      *
-     * @return number of files
+     * @return a  {@link IndexShardSnapshotStatus.Copy}
      */
-    public int numberOfFiles() {
-        return numberOfFiles;
+    public synchronized IndexShardSnapshotStatus.Copy asCopy() {
+        return new IndexShardSnapshotStatus.Copy(stage.get(), startTime, totalTime, numberOfFiles, processedFiles, totalSize, processedSize,
+                                                 indexVersion, failure);
     }
 
-    /**
-     * Total snapshot size
-     *
-     * @return snapshot size
-     */
-    public long totalSize() {
-        return totalSize;
+    public static IndexShardSnapshotStatus newInitializing() {
+        return new IndexShardSnapshotStatus(Stage.INIT, 0L, 0L, 0, 0, 0, 0, 0, null);
     }
 
-    /**
-     * Number of processed files
-     *
-     * @return number of processed files
-     */
-    public int processedFiles() {
-        return processedFiles;
+    public static IndexShardSnapshotStatus newFailed(final String failure) {
+        if (failure == null) {
```

also add an assertion here so we fail hard in our tests?